astropy/coordinates/sky_coordinate.py

```diff
@@ -65,19 +66,35 @@ def _repr_data(self):
         return repr_data
 
     def _represent_as_dict(self):
-        obj = self._parent
-        attrs = (list(obj.representation_component_names) +
-                 list(frame_transform_graph.frame_attributes.keys()))
+        sc = self._parent
+        attrs = list(sc.representation_component_names)
+
+        # Don't output distance unless it's actually distance.
+        if issubclass(sc.data.__class__, UnitSphericalRepresentation):
+            attrs = attrs[:-1]
+
+        diff = sc.data.differentials.get('s')
+        if diff:
+            diff_attrs = list(sc.get_representation_component_names('s'))
+            # Don't output proper motions if they haven't been specified.
+            if issubclass(diff.base_representation, RadialRepresentation):
+                diff_attrs = diff_attrs[2:]
+            # Don't output radial velocity unless it's actually velocity.
+            elif issubclass(diff.base_representation,
+                            UnitSphericalRepresentation):
+                diff_attrs = diff_attrs[:-1]
+            attrs.extend(diff_attrs)
+
+        if getattr(sc, 'obstime'):
```

Comment:
I'm not sure why this is needed: `obstime` is a frame attribute, so it should already be included below.

Reply:
It was indeed unnecessary.